accord-core/src/main/java/accord/primitives/TxnId.java

```diff
@@ -181,7 +181,7 @@ public static TxnId minForEpoch(long epoch)
         return new TxnId(epochMsb(epoch), 0, Id.NONE);
     }
 
-    private static final Pattern PARSE = Pattern.compile("\\[(?<epoch>[0-9]+),(?<hlc>[0-9]+),(?<flags>[0-9]+)\\([KR][REWSXL]\\),(?<node>[0-9]+)]");
+    private static final Pattern PARSE = Pattern.compile("\\[(?<epoch>[0-9]+),(?<hlc>[0-9]+),(?<flags>[0-9]+)\\([KR][REWSXL]\\),(?<node>-?[0-9]+)]");
```

Comment:
TCM allows negative nodes (and epoch)... so extended the pattern to better support TCM (though this doesn't support negative epochs... those are for transitioning from Gossip)

Reply:
Accord doesn’t, so we shouldn’t support that gee without a lot of wider work to make sure it is safe.

in fact epochs alone should immediately break as we truncate to only 48 bits.

Reply:
I think in principle negative node ids may be fine, but I would prefer not to find out if possible, and would prefer anyway to encode them as positive values (treat as unsigned ints)

Reply:
we spoke in slack but confirmed negative nodes isn't a thing in TCM.  I was getting confused about the migration case as `host_id: UUID` is based off `UUID.random` and the conversion logic to `int` would have allowed negatives... but the `host_id != nodeId.asUUID`, so looks like I was just confused